webpush.go

```diff
@@ -162,7 +161,7 @@ func SendNotification(message []byte, s *Subscription, options *Options) (*http.
 	ciphertext := gcm.Seal([]byte{}, nonce, plaintext, nil)
 
 	// POST request
-	req, err := http.NewRequest("POST", s.Endpoint, ioutil.NopCloser(bytes.NewReader(ciphertext)))
+	req, err := http.NewRequest("POST", s.Endpoint, bytes.NewReader(ciphertext))
```

Comment:
I could also have fixed this by doing 

```
req, err := http.NewRequest("POST", s.Endpoint, nil)
req.Body = ioutil.NopCloser(bytes.NewReader(ciphertext))
req.ContentLength = int64(len(ciphertext))
```

[which is what gauntface's implementation does](https://github.com/gauntface/web-push-go/blob/master/webpush/push.go#L64), but it didn't seem like the `ioutil.NopCloser` was actually getting us anything, so I just removed it instead which also allows ContentLength to be set, consistent with the docs at the top of `http.NewRequest`:

> // NewRequest returns a new Request given a method, URL, and optional body.
//
// If the provided body is also an io.Closer, the returned
// Request.Body is set to body and will be closed by the Client
// methods Do, Post, and PostForm, and Transport.RoundTrip.
//
// NewRequest returns a Request suitable for use with Client.Do or
// Transport.RoundTrip. To create a request for use with testing a
// Server Handler, either use the NewRequest function in the
// net/http/httptest package, use ReadRequest, or manually update the
// Request fields. See the Request type's documentation for the
// difference between inbound and outbound request fields.
//
// If body is of type *bytes.Buffer, *bytes.Reader, or
// *strings.Reader, the returned request's ContentLength is set to its
// exact value (instead of -1), GetBody is populated (so 307 and 308
// redirects can replay the body), and Body is set to NoBody if the
// ContentLength is 0.